Source/Laser/LaserProfilesImpl/LaserProfileFromFile.cpp

```diff
@@ -0,0 +1,570 @@
+/* Copyright 2019-2020 Luca Fedeli
+ * Ilian Kara-Mostefa
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#include "Laser/LaserProfiles.H"
+
+#include "Utils/Algorithms/LinearInterpolation.H"
+#include "Utils/Algorithms/UpperBound.H"
+#include "Utils/Parser/ParserUtils.H"
+#include "Utils/TextMsg.H"
+#include "Utils/WarpX_Complex.H"
+#include "Utils/WarpXConst.H"
+
+#include <ablastr/warn_manager/WarnManager.H>
+
+#include <AMReX.H>
+#include <AMReX_Algorithm.H>
+#include <AMReX_Config.H>
+#include <AMReX_Extension.H>
+#include <AMReX_GpuContainers.H>
+#include <AMReX_GpuDevice.H>
+#include <AMReX_GpuLaunch.H>
+#include <AMReX_GpuQualifiers.H>
+#include <AMReX_PODVector.H>
+#include <AMReX_ParallelDescriptor.H>
+#include <AMReX_ParmParse.H>
+#include <AMReX_Print.H>
+#include <AMReX_REAL.H>
+#include <AMReX_Vector.H>
+
+#include <algorithm>
+#include <cmath>
+#include <cstdint>
+#include <fstream>
+#include <iterator>
+#include <limits>
+#include <string>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+// example: data handling & print
+#include <vector>   // std::vector
+#include <iostream> // std::cout
+#include <memory>   // std::shared_ptr
+
+#ifdef WARPX_USE_OPENPMD
+#   include <openPMD/openPMD.hpp>
+    namespace io = openPMD;
+#endif
+
+using namespace amrex;
+
+void
+WarpXLaserProfiles::FromFileLaserProfile::init (
+    const amrex::ParmParse& ppl,
+    CommonLaserParameters params)
+{
+    if (!std::numeric_limits< double >::is_iec559)
+    {
+        ablastr::warn_manager::WMRecordWarning("Laser",
+            "(Double does not comply with IEEE 754: bad"
+            "things will happen parsing the X, Y and T profiles for the laser!)",
+            ablastr::warn_manager::WarnPriority::high);
+    }
+    // Parse the lasy or binary file
+    ppl.query("lasy_file_name", m_params.lasy_file_name);
+    ppl.query("binary_file_name", m_params.binary_file_name);
+    std::string lasy_file_name = m_params.lasy_file_name;
+    std::string binary_file_name = m_params.binary_file_name;
+    m_params.file_in_lasy_format = false;
+    if (!lasy_file_name.empty() && binary_file_name.empty()) {
+#ifdef WARPX_USE_OPENPMD
+        m_params.file_in_lasy_format = true;
+        parse_lasy_file(lasy_file_name);
+#else
+    amrex::Abort(Utils::TextMsg::Err("WarpX has to be compiled with the option openPMD=ON to read a lasy file"));
+    amrex::ignore_unused(ppl, params);
+#endif
+    } else if (lasy_file_name.empty() && !binary_file_name.empty()){
+        parse_binary_file(binary_file_name);
+    }
+    else if (!lasy_file_name.empty() && !binary_file_name.empty()){
+        Abort("Specify only one file format to load, either lasy_file_name or binary_file_name");
```

Comment:
I would suggest using the new macro:
```suggestion
        WARPX_ABORT_WITH_MESSAGE("Specify only one file format to load, either lasy_file_name or binary_file_name");
```